python/pylibcudf/pylibcudf/libcudf/wrappers/timestamps.pxd

```diff
@@ -1,11 +1,31 @@
-# Copyright (c) 2020-2024, NVIDIA CORPORATION.
+# Copyright (c) 2020-2025, NVIDIA CORPORATION.
+from libc.time cimport time_t
 
 from libc.stdint cimport int32_t, int64_t
+from pylibcudf.libcudf.wrappers.durations cimport (
+    duration_s,
+    duration_ms,
+    duration_us,
+    duration_ns,
+)
 
+cdef extern from "<chrono>" namespace "cuda::std::chrono" nogil:
+    cdef cppclass microseconds:
+        pass
+    cdef cppclass time_point[T, U]:
+        pass
+
+    cdef cppclass system_clock:
+        @staticmethod
+        time_point[system_clock, microseconds] from_time_t(time_t t) except +
```

Comment:
@vyasr could use some Cython help with this definition.

Ideally I want this to be able to return `time_point[system_clock, duration_s]`, `time_point[system_clock, duration_ms]`, `time_point[system_clock, duration_us]`, or `time_point[system_clock, duration_ns]` so when used below I could just do

```python
cdef time_point[system_clock, duration_reso] tp = system_clock.from_time_t(
    <time_t>(dt.timestamp() * factor)
)
(<timestamp_scalar[timestamp_reso]*>c_obj.get()).set_value(tp)
```

From the docs this just returns a `time_point[system_clock]`, but I'm not sure if I could define this in a way to return multiple durations https://en.cppreference.com/w/cpp/chrono/system_clock/from_time_t

Reply:
It seems like `system_clock::from_time_t` always uses the duration associated with the clock itself. You might have to [`time_point_cast`](https://en.cppreference.com/w/cpp/chrono/time_point/time_point_cast) the result afterwards.

Reply:
I was able to find a simpler method (`chrono::time_stamp` can be constructed from a `chrono::duration`) afterall so I didn't need this.

Reply:
Nice I'm glad that simplified things!